Add deprecation warnings for `send()`, `transfer()`, ABI coder v1, contract comparisons, virtual modifiers and `memory-safe-assembly` comment (#16174)
ethereum/solidity: docs/contracts/inheritance.rst

Reviewer: Maybe wouldn't hurt to clarify that this is deprecated already in the headline?

```suggestion
Modifier Overriding (deprecated)
```

    contract Base
    {

>         // This will report a warning

Reviewer: ```suggestion
        // This will emit a deprecation warning
```

    contract Base1
    {

>         // This will report a warning

Reviewer: ```suggestion
        // This will emit a deprecation warning
```

        modifier foo() virtual {_;}
    }

    contract Base2
    {

>         // This will report a warning

Reviewer: ```suggestion
        // This will emit a deprecation warning
```

        modifier foo() override(Base1, Base2) {_;}
    }

.. warning::

>     ``virtual`` modifiers are deprecated and scheduled.

Reviewer 2: ```suggestion
    ``virtual`` modifiers are deprecated and scheduled for removal.
```

>     virtual modifiers will be deprecated in the next breaking version (0.9).

Reviewer: ```suggestion
    ``virtual`` modifiers are deprecated and scheduled for removal in the next breaking version (0.9).
```
